Spec/RealtimeClientChannel.swift

```diff
@@ -1110,6 +1110,208 @@ class RealtimeClientChannel: QuickSpec {
 
                 }
 
+                // RTL6c
+                context("Connection state conditions") {
+
+                    // RTL6c1
+                    it("if the connection is CONNECTED and the channel is ATTACHED then the messages should be published immediately") {
+                        let client = AblyTests.newRealtime(AblyTests.commonAppSetup())
+                        defer { client.close() }
+                        let channel = client.channels.get("test")
+                        channel.attach()
+
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+                        expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.publish(nil, data: "message") { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                            expect((client.transport as! TestProxyTransport).protocolMessagesSent.filter({ $0.action == .Message })).to(haveCount(1))
+                        }
+                    }
+
+                    // RTL6c2
+                    context("the message should be queued and delivered as soon as the connection state returns to CONNECTED if the connection is") {
```

Comment:
The tests in this context test `the message should be queued`, but don't test `and delivered as soon as the connection state returns to CONNECTED`.  Should we?